presto-pinot/presto-pinot-lib/src/main/java/com/facebook/presto/pinot/PinotConfig.java

```diff
@@ -0,0 +1,400 @@
+import javax.validation.constraints.NotNull;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class PinotConfig
+{
+    public static final int DEFAULT_LIMIT_LARGE_FOR_SEGMENT = Integer.MAX_VALUE;
+    public static final int DEFAULT_MAX_BACKLOG_PER_SERVER = 30;
+    public static final int DEFAULT_MAX_CONNECTIONS_PER_SERVER = 30;
+    public static final int DEFAULT_MIN_CONNECTIONS_PER_SERVER = 10;
+    public static final int DEFAULT_THREAD_POOL_SIZE = 30;
+    public static final int DEFAULT_NON_AGGREGATE_LIMIT_FOR_BROKER_QUERIES = 25_000;
+
+    // There is a perf penalty of having a large topN since the structures are allocated to this size
+    // So size this judiciously
+    public static final int DEFAULT_TOPN_LARGE = 10_000;
+
+    private static final long DEFAULT_IDLE_TIMEOUT_MINUTE = 5L; // 5 minutes
+    private static final long DEFAULT_CONNECTION_TIMEOUT_MINUTE = 1L; // 1 minute
+    private static final int DEFAULT_ESTIMATED_SIZE_IN_BYTES_FOR_NON_NUMERIC_COLUMN = 20;
+
+    private int maxConnectionsPerServer = DEFAULT_MAX_CONNECTIONS_PER_SERVER;
+    private String controllerRestService;
+    private String serviceHeaderParam = "RPC-Service";
+    private String callerHeaderValue = "presto";
+    private String callerHeaderParam = "RPC-Caller";
+
+    private String controllerUrl;
+    private String restProxyUrl;
+    private String restProxyServiceForQuery;
+
+    private int limitLargeForSegment = DEFAULT_LIMIT_LARGE_FOR_SEGMENT;
+    private int topNLarge = DEFAULT_TOPN_LARGE;
+
+    private Duration idleTimeout = new Duration(DEFAULT_IDLE_TIMEOUT_MINUTE, TimeUnit.MINUTES);
+    private Duration connectionTimeout = new Duration(DEFAULT_CONNECTION_TIMEOUT_MINUTE, TimeUnit.MINUTES);
+
+    private int threadPoolSize = DEFAULT_THREAD_POOL_SIZE;
+    private int minConnectionsPerServer = DEFAULT_MIN_CONNECTIONS_PER_SERVER;
+    private int maxBacklogPerServer = DEFAULT_MAX_BACKLOG_PER_SERVER;
+    private int estimatedSizeInBytesForNonNumericColumn = DEFAULT_ESTIMATED_SIZE_IN_BYTES_FOR_NON_NUMERIC_COLUMN;
+    private Map<String, String> extraHttpHeaders = ImmutableMap.of();
+    private Duration metadataCacheExpiry = new Duration(2, TimeUnit.MINUTES);
+
+    private boolean allowMultipleAggregations;
+    private boolean scanParallelismEnabled = true;
+    private int numSegmentsPerSplit = 1;
+    private boolean ignoreEmptyResponses;
+    private int fetchRetryCount = 2;
+    private boolean usePrestoDateTrunc;
+    private int nonAggregateLimitForBrokerQueries = DEFAULT_NON_AGGREGATE_LIMIT_FOR_BROKER_QUERIES;
+
+    @NotNull
+    public Map<String, String> getExtraHttpHeaders()
+    {
+        return extraHttpHeaders;
+    }
+
+    @Config("pinot.extra-http-headers")
+    public PinotConfig setExtraHttpHeaders(String headers)
+    {
+        extraHttpHeaders = ImmutableMap.copyOf(Splitter.on(",").trimResults().omitEmptyStrings().withKeyValueSeparator(":").split(headers));
```

Comment:
Make 
```
Splitter.on(",").trimResults().omitEmptyStrings().withKeyValueSeparator(":")
```
a private static final variable.

Reply:
It seems like an overkill for a one time use method. This code shouldn't be called in the critical path for any definition of critical path.